core/domain/platform_parameter_domain.py

```diff
@@ -17,19 +17,59 @@
 """Domain objects for platform parameters."""
 
 from __future__ import annotations
-
 import json
 import re
 
+import enum as enum
+from typing import Any, Dict, List, Optional, Tuple
+from typing_extensions import TypedDict
+
 from core import feconf
 from core import python_utils
 from core import utils
 from core.constants import constants
 from core.domain import change_domain
 
-SERVER_MODES = python_utils.create_enum('dev', 'test', 'prod') # pylint: disable=invalid-name
+
+class SERVER_MODES(enum.Enum):
+    dev = 'dev'
+    test = 'test'
+    prod = 'prod'
+
+
+class DATA_TYPES(enum.Enum):
+    bool = 'bool'
+    string = 'string'
+    number = 'number'
+
+
+class ServerContextDict(TypedDict):
+    server_mode: SERVER_MODES
+
+
+class FilterDict(TypedDict):
+    type: str
+    conditions: List[Tuple[str, str]]
+
+
+class PlatformParameterRuleDict(TypedDict):
+    filters: List[FilterDict]
+    # value_when_matched is * so its type is Any
```

Comment:
This is not a proper explanation, you should explain why conceptually it is `Any`.

Reply:
done